init.ts

```diff
@@ -0,0 +1,231 @@
+import _ from 'lodash';
+
+const paginationSchemas = {
+  Meta: {
+    properties: {
+      totalResults: {
+        type: 'integer',
+        description: 'Total number of results',
+        example: 10,
+      },
+      totalPages: {
+        type: 'integer',
+        description: 'Total number of pages',
+        example: 10,
+      },
+      currentPageNumber: {
+        type: 'integer',
+        description: 'Page number of the returned results',
+        example: 1,
+      },
+      currentPageSize: {
+        type: 'integer',
+        description: 'Number of results per page',
+        example: 25,
+      },
+    },
+  },
+  PaginationLinks: {
+    properties: {
+      first: {
+        type: 'string',
+        format: 'uri',
+        description: 'The first page of data',
+      },
+      last: {
+        type: 'string',
+        format: 'uri',
+        description: 'The last page of data',
+      },
+      prev: {
+        type: 'string',
+        format: 'uri',
+        description: 'The previous page of data',
+      },
+      next: {
+        type: 'string',
+        format: 'uri',
+        description: 'The next page of data',
+      },
+    },
+  },
+};
+
+const paginationParameters = {
+  pageNumber: {
+    name: 'page[number]',
+    in: 'query',
+    required: false,
+    description: 'Page number of results',
+    schema: {
+      type: 'integer',
+      minimum: 1,
+      default: 1,
+    },
+  },
+  pageSize: {
+    name: 'page[size]',
+    in: 'query',
+    required: false,
+    description: 'Number of results to return',
+    schema: {
+      type: 'integer',
+      minimum: 1,
+      maximum: 500,
+      default: 25,
+    },
+  },
+};
+
+const errorSchema = {
+  content: {
+    'application/json': {
+      schema: {
+        $ref: '#/components/schemas/ErrorResult',
+      },
+    },
+  },
+};
+
+/**
+ * Creates the initial openapi object template
+ *
+ * @param config - The config file object
+ * @returns The updated openapi object
+ */
+const init = (config: any) => {
+  const containsPaginated = _.some(config.resources, (resource) => resource.paginate);
+  const extraSchemas = containsPaginated ? paginationSchemas : {};
+  const extraParameters = containsPaginated ? paginationParameters : {};
+
+  const openapi = {
+    openapi: '3.0.0',
+    info: {
+      title: config.title,
+      description: 'REPLACEME',
+      version: config.version,
+      license: {
+        name: 'REPLACEME',
+        url: 'REPLACEME',
+      },
+      contact: {
+        name: 'IS Data Architecture Team',
+        url: 'https://is.oregonstate.edu/data-architecture',
+        email: 'isdataarchitecture@oregonstate.edu',
+      },
+    },
+    externalDocs: {
+      description: 'GitHub Repository',
+      url: config.githubUrl,
+    },
+    servers: [
+      {
+        url: `https://api.oregonstate.edu/${config.version}`,
```

Comment:
I'm not a big fan of destructuring because I like being able to see in line where a variable is coming from. Since we are already using so much from `config` I think it's fine to leave as is.